packages/flutter/test/material/app_bar_test.dart

```diff
@@ -1422,4 +1422,26 @@ void main() {
       statusBarIconBrightness: Brightness.dark,
     ));
   });
+
+  testWidgets('AppBar with shape', (WidgetTester tester) async {
+    const RoundedRectangleBorder roundedRectangleBorder = RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(15.0)));
+    await tester.pumpWidget(
+      MaterialApp(
+        home: AppBar(
+          leading: const Text('L'),
+          title: const Text('No Scaffold'),
+          shape: roundedRectangleBorder,
+          actions: const <Widget>[Text('A1'), Text('A2')],
+        ),
+      ),
+    );
+
+    final Finder appBarFinder = find.byType(AppBar);
+
+    AppBar getAppBarWidget() {
+      return tester.widget<AppBar>(appBarFinder);
+    }
+
+    expect(getAppBarWidget().shape, roundedRectangleBorder);
```

Comment:
Checking that the AppBar's shape parameter correctly initializes its shape property is useful, however you should also verify that the the AppBar Material widget's shape property is set.

You may find https://docs.flutter.io/flutter/flutter_test/CommonFinders/descendant.html to be useful for this.